File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    largest = 0
    other_size = 0
    for size in sizes:
        if largest < max(size):
            largest = max(size)
        if other_size < min(size):
            other_size = min(size)
    
    answer = largest * other_size
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let horizon = sizes[0][0];
    let vertical = sizes[0][1];
    for (size of sizes) {
        if (size[0] > horizon || size[1] > vertical) {
            if (Math.max(horizon, size[0]) * Math.max(vertical, size[1]) < Math.max(horizon, size[1]) * Math.max(vertical, size[0])) {
                horizon = Math.max(horizon, size[0]);
                vertical = Math.max(vertical, size[1]);
            } else {
                horizon = Math.max(horizon, size[1]);
                vertical = Math.max(vertical, size[0]);
            }
        }
    }
    return horizon * vertical;
}